guides/real_time_speech_recognition.md

```diff
@@ -0,0 +1,231 @@
+* Add the `"state"` components to the `inputs` and `outputs` in `Interface`
+
+Here's what the code looks like:
+
+```python
+import gradio as gr
+
+def transcribe(audio, state=""):
+    text = p(audio)["text"]
+    state += text + " "
+    return state, state
+
+gr.Interface(
+    fn=transcribe, 
+    inputs=[
+        gr.inputs.Audio(source="microphone", type="filepath"), 
+        "state"
+    ],
+    outputs=[
+        "textbox",
+        "state"
+    ],
+    live=True).launch()
+```
+
+Notice that we've also made one other change, which is that we've set `live=True`. This keeps the Gradio interface running constantly, so it automatically transcribes audio without the user having to repeatedly hit the submit button.
+
+Let's see how it does!
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/streaming-asr/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
+
+One thing that you may notice is that the transcription quality has dropped since the chunks of audio are so small, they lack the context to properly be transcribed. A "hacky" fix to this is to simply increase the runtime of the `transcribe()` function so that longer audio chunks are processed. We can do this by adding a `time.sleep()` inside the function, as shown below (we'll see a proper fix next) 
+
+```python
+from transformers import pipeline
+import gradio as gr
+import time
+
+p = pipeline("automatic-speech-recognition")
+
+def transcribe(audio, state=""):
+    time.sleep(2)
+    text = p(audio)["text"]
+    state += text + " "
+    return state, state
+
+gr.Interface(
+    fn=transcribe, 
+    inputs=[
+        gr.inputs.Audio(source="microphone", type="filepath"), 
+        "state"
+    ],
+    outputs=[
+        "textbox",
+        "state"
+    ],
+    live=True).launch()
+```
+
+Try the demo below to see the difference!
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/streaming-asr-paused/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```

Comment:
this keeps erroring out for me in the guide, but not in the actual space. not sure what is going on, might just be something i'm doing so if it works for you you can ignore.